javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -556,4 +559,22 @@ public static Name getElementValueClassName(AnnotationMirror anno, CharSequence
         }
         map.put(key, Collections.unmodifiableSet(result));
     }
+
+    /**
+     * Returns the annotations explicitly written on a constructor result.
+     * Callers should check that {@code constructorDeclaration} is in fact a declaration
+     * of a constructor.
+     *
+     * @param constructorDeclaration Declaration tree of constructor
+     * @return set of annotations explicit on the resulting type of the constructor
+     */
+    public static Set<AnnotationMirror> getExplicitAnnotationsOnConstructorResult(MethodTree constructorDeclaration) {
```

Comment:
I would prefer to leave it in AnnotationUtils where it is more likely to be discovered and used.  I also think when `@TargetLocations` are enforced (Issue #515), we will want to add methods to get explicit annotations from other trees.